k8s-test-runner/runner/runner.go

```diff
@@ -0,0 +1,217 @@
+	ErrNoJobs      = errors.New("HelmValues should contain \"jobs\" field used to scale your cluster jobs, jobs must be > 0")
+)
+
+// Config defines k8s jobs settings
+type Config struct {
+	ChartPath            string
+	Namespace            string
+	KeepJobs             bool
+	DockerCmdExecPath    string
+	DockerfilePath       string
+	DockerIgnoreFilePath string
+	BuildScriptPath      string
+	BuildCtxPath         string
+	ImageTag             string
+	RegistryName         string
+	RepoName             string
+	HelmDeployTimeoutSec string
+	HelmValues           map[string]string
+	// generated values
+	tmpHelmFilePath string
+}
+
+func (m *Config) Defaults() error {
+	m.HelmValues["namespace"] = m.Namespace
+	// nolint
+	m.HelmValues["sync"] = fmt.Sprintf("a%s", uuid.NewString()[0:5])
+	if m.HelmDeployTimeoutSec == "" {
+		m.HelmDeployTimeoutSec = defaultHelmDeployTimeoutSec
+	}
+	if m.HelmValues["test.timeout"] == "" {
+		m.HelmValues["test.timeout"] = "12h"
+	}
+	if m.HelmValues["resources.requests.cpu"] == "" {
+		m.HelmValues["resources.requests.cpu"] = DefaultRequestsCPU
+	}
+	if m.HelmValues["resources.requests.memory"] == "" {
+		m.HelmValues["resources.requests.memory"] = DefaultRequestsMemory
+	}
+	if m.HelmValues["resources.limits.cpu"] == "" {
+		m.HelmValues["resources.limits.cpu"] = DefaultLimitsCPU
+	}
+	if m.HelmValues["resources.limits.memory"] == "" {
+		m.HelmValues["resources.limits.memory"] = DefaultLimitsMemory
+	}
+	return nil
+}
+
+func (m *Config) Validate() (err error) {
+	if m.Namespace == "" {
+		err = errors.Join(err, ErrNoNamespace)
+	}
+	if m.HelmValues["jobs"] == "" {
+		err = errors.Join(err, ErrNoJobs)
+	}
+	return
+}
+
+type K8sTestRun struct {
+	cfg    *Config
+	c      *k8s_client.Client
+	Ctx    context.Context
+	Cancel context.CancelFunc
+}
+
+func NewK8sTestRun(cfg *Config) (*K8sTestRun, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+	if err := cfg.Defaults(); err != nil {
+		return nil, err
+	}
+	log.Info().Interface("Config", cfg).Msg("Cluster configuration")
+	dur, err := time.ParseDuration(cfg.HelmValues["test.timeout"])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse test timeout duration")
+	}
+	ctx, cancelFunc := context.WithTimeout(context.Background(), dur)
+	cp := &K8sTestRun{
+		cfg:    cfg,
+		c:      k8s_client.NewClient(),
+		Ctx:    ctx,
+		Cancel: cancelFunc,
+	}
+	return cp, nil
+}
+
+func (m *K8sTestRun) getChartPath() string {
+	if m.cfg.ChartPath == "" {
+		// Use default chart
+		_, f, _, _ := runtime.Caller(0)
+		dir := path.Join(path.Dir(f))
+		chartPath := path.Join(dir, "../chart")
+		return chartPath
+	}
+	return m.cfg.ChartPath
+}
+
+func (m *K8sTestRun) deployHelm(testName string) error {
+	deleteCmd := fmt.Sprintf("helm delete %s -n %s --timeout 3m", testName, m.cfg.Namespace)
+	log.Info().Str("Cmd", deleteCmd).Msg("Delete previous release if exists") // This is needed when running on CI with GAP
+	ExecCmd(deleteCmd)
```

Comment:
This should be removed before we merge, otherwise, we can fail with an error when fixed.

Reply:
@skudasov yep, I re-tested and it works fine with latest GAP. I removed it